x-pack/plugins/maps/public/classes/styles/vector/properties/dynamic_style_property.tsx

```diff
@@ -122,6 +122,10 @@ export class DynamicStyleProperty<T extends object>
     this._getFieldFormatter = getFieldFormatter;
   }
 
+  getStyleMetaDataRequest() {
```

Comment:
styles can have 2 types of sources, feature source and join source. This means the request id can not be hard coded to `SOURCE_META_DATA_REQUEST_ID`. Luckily this logic already exists and you can use `_getStyleMetaDataRequestId` to get the request id. 

```
getStyleMetaDataRequest() {
    const dataRequestId = this._getStyleMetaDataRequestId(this.getFieldName());
    return dataRequestId
      ? this._layer.getDataRequest(dataRequestId)
      : undefined;
}|
```

There is some code duplicate where `_getStyleMetaDataRequestId` is currently being used. Would you mind replacing `_getStyleMetaDataRequestId` usage with `getStyleMetaDataRequest`?